.github/workflows/forge-fmt.yml

```diff
@@ -0,0 +1,26 @@
+name: Forge format
+
+on:
+  workflow_call:
+    secrets:
+      MACHINE_USER_PRIVATE_KEY:
+        required: true
+
+jobs:
+  forge-fmt:
+    name: Formatting
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+          ssh-key: ${{ secrets.MACHINE_USER_PRIVATE_KEY }}
+
+      - name: Install Foundry
+        uses: onbjerg/foundry-toolchain@v1
+        with:
+          version: nightly
+
+      - name: Check formatting
+        run: forge fmt --check
```

Comment:
Let's put a newline at the end of files please (there is a vscode setting called "FIles: Insert Final Newline" that allows you to never have to think about that again)